docs/threat-model.md

```diff
@@ -0,0 +1,158 @@
+privileges that they should not have access to, as described in (2).
+
+### (2) Privilege Escalation
+
+As there will be a hierarchy of users with varying capabilities, it should not be possible for a
+user to gain additional privileges without them being granted by an administrator. Violation of
+this assumption would allow ordinary users to take advantage of privileges they should not have
+access to, potentially wreaking havoc regardless of malicious intent.
+
+### (3) Direct Access to Backend Services
+
+Backend services should not be directly accessible. Violation of this principle could permit actors
+to access APIs they should not have access to, potentially wreaking havoc on internal state or
+exposing data that should remain private to them.
+
+### (4) API Gateway Overloading
+
+As the API Gateway is the main way for users to interact with applications, it serves as a single
+point of failure for the entire application. By overloading the API Gateway, malicious actors (or
+just a sudden spike of traffic) can disrupt the availability of the application for all users.
+
+### (5) Network Eavesdropping
+
+There may be actors that are listening in on communications between our users or between services 
+in our application. Through eavesdropping on network communications, malicious actors could learn
+private user/system/operational data that they should not have access to. Further, they may be able
+to construct spoofed requests for abuse in conjunction with Threat (3).
+
+## Defense Mechanisms
+
+### (1, 2) User Authentication
+
+To protect against user impersonation, the application will utilize mechanisms of user authentication
+to validate the user's identity prior to allowing access to any privileged operations or data (including
+operations or data tied to user identity).
+
+#### Password Authentication
+
+Upon user creation, users will select a password to use for authentication. We will impose several
+requirements to ensure the password is not trivially bruteforceable.
+
+1. Minimum of 12 characters; maximum of 60 characters
+2. Includes at least one number
+3. Includes at least one uppercase and lowercase alphabetic character
+4. Includes at least one non-alphabetic character (symbol)
+
+These passwords will be stored in a User Authentication service. They will be salted and hashed to
+protect against offline attacks if the password database were ever leaked.
+
+* The salt will be a cryptographically random string.
+* The chosen hash function will be computationally expensive to deter offline attacks.
+
+TODO: pick hash function. `bcrypt` seems preferred nowadays? it may also handle salting?
+
+#### Two Factor Authentication (2FA)
+
+As user passwords may be shared between multiple services, may be bruteforceable, may be leaked 
+through a variety of means, etc., our application will utilize 2FA to further validate user
+identities.
+
+TODO: mechanism? email?
+
+### (1, 2) User Sessions
```

Comment:
There are some special users in the system. Specifically users who are created to access the API from game mods, e.g. within `potato_aar`. These users would ideally have special means by either a) bypassing user sessions b) having infinite user sessions c) having quicker authentication

Reply:
i guess the concern is over one specific bot, the one that sends data from arma to processor. with the currently implementation is does not receive any data and instead just spits data through a udp connection for simplicity. this can be altered, i may want to do websockets instead, but the use case stands. if you want to go “no this isn’t possible” that’s fine as well

Reply:
This has been agreed to be discussed outside of this PR.